src/Controller/Component/AuthenticationComponent.php

```diff
@@ -89,6 +109,47 @@ public function beforeFilter()
         ], $this->getController());
     }
 
+    /**
+     * Start up event handler
+     *
+     * If `requireIdentity` is true, the action will be compared to the
+     * allowed actions.
+     *
+     * @return void
+     */
+    public function startup()
+    {
+        if (!$this->getConfig('requireIdentity')) {
+            return;
+        }
+        $request = $this->getController()->request;
+        $action = $request->getParam('action');
+        if (in_array($action, $this->unauthenticatedActions)) {
+            return;
+        }
+
+        $identity = $request->getAttribute('identity');
+        if (!$identity) {
+            throw new UnauthorizedException([]);
```

Comment:
Should it throw MissingIdentityException?

Reply:
Or `UnauthenticatedException` - but MissingIdentityException as generic one can also be fine.

Reply:
With a `loginRedirect` or `loginAction` present, wouldn't it make sense to redirect to login then instead from the component?
The exception is only useful for stateless (api) access, right?

Reply:
Redirection is performed from a middleware using unauthorized handlers.

Reply:
@robertpustulka You're right, `MissingIdentity` would be better, as it is handled by the Unauthorized handlers.

Reply:
Should it be a new `UnauthenticatedException` ?
`UnauthorizedException` is used for http auth AFAIR

Reply:
At this point definitely not authorize, but authenticate, yes.

Reply:
I was attempting to reuse the [existing exception](https://github.com/cakephp/authentication/blob/master/src/Authenticator/UnauthorizedException.php) which generates a 401 error, which is what I think we want here too.